tests/sentry/integrations/github_enterprise/test_client.py

```diff
@@ -0,0 +1,154 @@
+    "filepath": "CODEOWNERS",
+    "html_url": "https://github.example.org/Test-Organization/foo/blob/master/CODEOWNERS",
+    "raw": "docs/*    @jianyuan   @getsentry/ecosystem\n* @jianyuan\n",
+}
+
+
+class GitHubAppsClientTest(TestCase):
+    base_url = "https://github.example.org/api/v3"
+
+    def setUp(self):
+        super().setUp()
+
+        patcher_1 = mock.patch(
+            "sentry.integrations.github_enterprise.client.get_jwt", return_value="jwt_token_1"
+        )
+        patcher_1.start()
+        self.addCleanup(patcher_1.stop)
+
+        patcher_2 = mock.patch(
+            "sentry.integrations.github_enterprise.integration.get_jwt", return_value="jwt_token_1"
+        )
+        patcher_2.start()
+        self.addCleanup(patcher_2.stop)
+
+        integration = self.create_integration(
+            organization=self.organization,
+            provider="github_enterprise",
+            name="Github Test Org",
+            external_id="1",
+            metadata={
+                "access_token": None,
+                "expires_at": None,
+                "icon": "https://github.example.org/avatar.png",
+                "domain_name": "github.example.org/Test-Organization",
+                "account_type": "Organization",
+                "installation_id": "install_id_1",
+                "installation": {
+                    "client_id": "client_id",
+                    "client_secret": "client_secret",
+                    "id": "2",
+                    "name": "test-app",
+                    "private_key": "private_key",
+                    "url": "github.example.org",
+                    "webhook_secret": "webhook_secret",
+                    "verify_ssl": True,
+                },
+            },
+        )
+        self.repo = Repository.objects.create(
+            organization_id=self.organization.id,
+            name="Test-Organization/foo",
+            url="https://github.example.org/Test-Organization/foo",
+            provider="integrations:github_enterprise",
+            external_id=123,
+            integration_id=integration.id,
+        )
+        self.install = integration.get_installation(organization_id=self.organization.id)
+        self.client = self.install.get_client()
+        responses.add(
+            method=responses.POST,
+            url=f"{self.base_url}/app/installations/install_id_1/access_tokens",
+            body='{"token": "12345token", "expires_at": "2030-01-01T00:00:00Z"}',
+            status=200,
+            content_type="application/json",
+        )
+
+    @responses.activate
+    def test_check_file(self):
+        path = "src/sentry/integrations/github/client.py"
+        version = "master"
+        url = f"{self.base_url}/repos/{self.repo.name}/contents/{path}?ref={version}"
+
+        responses.add(
+            method=responses.HEAD,
+            url=url,
+            json={"text": 200},
+        )
+
+        assert hasattr(self.client, "check_file")
```

Comment:
this should instead check the type of `self.client` -- narrowing to the specific type that you're testing

Reply:
Fixed: I had to rename `self.client` to something else since `self.client` is already defined in `sentry.testutils.cases.TestCase`.

Reply:
lg2m

Reply:
@asottile-sentry I did some further testing. It seems that performing type narrowing in the `setUp()` method using `isinstance()` only doesn't carry the narrowed types into the test methods.

I had to use `cast()` during the assignment.

Let me know what's your preference.

Sample: https://mypy-play.net/?mypy=latest&python=3.11&flags=strict&gist=2bfe10edd440e6be3927d42e8c1c2f9e

```python
import unittest
from typing import cast


class Wide:
    pass


class Narrow(Wide):
    def hello(self) -> str:
        return "world"


def get_wide() -> Wide:
    return cast(Wide, Narrow())


# 1
class TestNarrow1(unittest.TestCase):
    def setUp(self) -> None:
        self.install = get_wide()
        assert isinstance(self.install, Narrow)

    def test_hello(self) -> None:
        # error: "Wide" has no attribute "hello"
        assert self.install.hello()


# 2
class TestNarrow2(unittest.TestCase):
    def setUp(self) -> None:
        # added cast()
        self.install = cast(Narrow, get_wide())
        assert isinstance(self.install, Narrow)

    def test_hello(self) -> None:
        # works
        assert self.install.hello()
```

Reply:
yeah it would only do that in `__init__`

`cast` should really only be used for mypy bugs -- but here's a pattern that should work for you without `cast`:

```python
x = get_wide()
assert isinstance(x, Narrow)
self.x = x
```

Reply:
Great tip! I have adjusted my PR accordingly.